Geometry/HGCalCommonData/src/HGCalDDDConstants.cc

```diff
@@ -2098,7 +2116,8 @@ int HGCalDDDConstants::waferType(int layer, int waferU, int waferV, bool fromFil
 }
 
 std::tuple<int, int, int> HGCalDDDConstants::waferType(HGCSiliconDetId const& id, bool fromFile) const {
-  const auto& index = HGCalWaferIndex::waferIndex(id.layer(), id.waferU(), id.waferV());
+  int waferU = (id.zside() > 0) ? -id.waferU() : id.waferU();
+  const auto& index = HGCalWaferIndex::waferIndex(id.layer(), waferU, id.waferV());
```

Comment:
here we need std..abs(waferU) or not?

Reply:
No - that wont work. Depending on side, one has to invert or not waferU. It does not need to take absolute value of waferU.